lib/boss_rds.py

```diff
@@ -0,0 +1,213 @@
+# Copyright 2019 The Johns Hopkins University Applied Physics Laboratory
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from contextlib import contextmanager
+from collections import namedtuple
+from mysql import connector
+import logging
+
+from lib.external import ExternalCalls
+from lib.names import AWSNames
+from lib import aws
+
+@contextmanager
+def connect_rds(session, domain):
+    """
+    Context manager with established connection to rds
+
+    Args:
+        session (Session): Open boto3 session.
+        domain (str): VPC such as integration.boss.
+
+    Returns:
+        sql: connection context
+    """
+    mysql_params = get_mysql_params(session, domain)
+
+    keypair = aws.keypair_lookup(session)
+    call = ExternalCalls(session, keypair, domain)
+
+    logging.info('Tunneling to DB (slow) . . .')
+    with call.tunnel(mysql_params.host, mysql_params.port, 'rds') as local_port:
+        try:
+            sql = connector.connect(
+                user=mysql_params.user, password=mysql_params.password, 
+                port=local_port, database=mysql_params.db
+            )
+            yield sql
+        finally:
+            sql.close()
+
+def get_mysql_params(session, domain):
```

Comment:
If you merge this functionality into the `connect_rds` function then you won't create two `ExternalCalls` objects, which when created query AWS for bastion and vault information.

Reply:
Good point!